module/resources.go

```diff
@@ -0,0 +1,537 @@
+// and reconstructed. If a resource is successfully reconfigured in place, (nil, nil) will be
+// returned.
+func (m *Module) reconfigureResource(
+	ctx context.Context, deps resource.Dependencies, conf *resource.Config, logLevel *logging.Level,
+) (resource.Resource, error) {
+	m.registerMu.Lock()
+	coll, ok := m.collections[conf.API]
+	if !ok {
+		m.registerMu.Unlock()
+		return nil, fmt.Errorf("no rpc service for %+v", conf)
+	}
+
+	res, err := coll.Resource(conf.ResourceName().Name)
+	if err != nil {
+		m.registerMu.Unlock()
+		return nil, err
+	}
+
+	resLogger, hasLogger := m.resLoggers[res]
+	m.registerMu.Unlock()
+	if hasLogger && logLevel != nil {
+		resLogger.SetLevel(*logLevel)
+	}
+
+	m.logger.Debugw("rebuilding", "name", conf.ResourceName().String())
+	err = res.Reconfigure(ctx, deps, *conf)
+	if err == nil {
+		return nil, nil
+	}
+
+	if !resource.IsMustRebuildError(err) {
+		return nil, err
+	}
+
+	if err := res.Close(ctx); err != nil {
+		m.logger.Error(err)
+	}
+
+	m.registerMu.Lock()
+	delete(m.activeResourceStreams, res.Name())
+	m.registerMu.Unlock()
+
+	resInfo, ok := resource.LookupRegistration(conf.API, conf.Model)
+	if !ok {
+		return nil, fmt.Errorf("resource with API %q and model %q not yet registered", conf.API, conf.Model)
+	}
+	if resInfo.Constructor == nil {
+		return nil, fmt.Errorf("invariant: no constructor for %q", conf.Model)
+	}
+
+	newRes, err := resInfo.Constructor(ctx, deps, *conf, m.logger)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := coll.ReplaceOne(conf.ResourceName(), newRes); err != nil {
+		m.registerMu.Unlock()
+		return nil, err
+	}
+
+	m.registerMu.Lock()
+	// We're modifying internal module maps now. We must not error out at this point without rolling
+	// back module state mutations.
+	m.logger.Infof("DBG. Deleting res from loggers: %p %v", res, res.Name())
+	m.logger.Infof("DBG. Added res to loggers: %p %v", newRes, newRes.Name())
+	delete(m.resLoggers, res)
+	m.resLoggers[newRes] = resLogger
+
+	if p, ok := newRes.(rtppassthrough.Source); ok {
+		m.streamSourceByName[res.Name()] = p
+	}
+
+	for _, resConfigureArgs := range m.internalDeps[res] {
+		m.logger.Infof("DBG. Chain reconfiguring. Initial: %v (%p -> %p) Dep: %v (%p)",
+			conf.ResourceName().Name, res, newRes, resConfigureArgs.toReconfig.Name().Name, resConfigureArgs.toReconfig)
+		deps, err := m.getDependenciesForConstruction(ctx, resConfigureArgs.depStrings)
+		if err != nil {
+			m.logger.Warn("Failed to get dependencies for reconfiguring dependent on parent rebuild",
+				"parent", resConfigureArgs.conf.ResourceName().Name,
+				"child", conf.ResourceName().Name,
+				"deps", resConfigureArgs.depStrings,
+				"err", err)
+			continue
+		}
+
+		m.registerMu.Unlock()
+		var nilLogLevel *logging.Level // pass in nil to avoid changing the log level
+		rebuiltRes, err := m.reconfigureResource(ctx, deps, resConfigureArgs.conf, nilLogLevel)
+		if err != nil {
+			m.logger.Warn("Failed to reconfigure parent on child rebuild",
```

Comment:
```suggestion
			m.logger.Warn("Failed to reconfigure dependent on parent rebuild",
```
to match wording in log a few lines above

Reply:
I made a different change to try and avoid parent/child altogether.